src/cmdhandler/command_handler.cpp

```diff
@@ -230,22 +272,45 @@ std::string command_handler::state_system(const std::vector<std::string>& tokens
         state_to_load.ParseFromIstream(&input);
 
         //clear robots state variable, fill from loaded State instance above
-        current_state.robots.clear();
+        current_state.robot.robots.clear();
         for(auto const &robot : state_to_load.robots()){
             std::vector<int> marker_ids_to_save;
             for(auto const &marker_id : robot.second.ids()){
                 marker_ids_to_save.push_back(marker_id);
             }
-            current_state.robots.insert(std::pair<std::string, std::vector<int>>(robot.first, marker_ids_to_save));
+            current_state.robot.robots.insert(std::pair<std::string, std::vector<int>>(robot.first, marker_ids_to_save));
         }
 
         //clear collector state variable, fill from loaded State
-        current_state.collectors.clear();
+        current_state.collector.collectors.clear();
         for(auto const &collector : state_to_load.collectors()){
             auto endpoint = asio::ip::udp::endpoint(
                     asio::ip::make_address(collector.second.address()),
                     collector.second.port());
-            current_state.collectors.insert(std::pair(collector.first, endpoint));
+            current_state.collector.collectors.insert(std::pair(collector.first, endpoint));
+        }
+
+        //fill url variable from loaded state
+        current_state.camera.url = state_to_load.url();
+
+        //clear camera_matrix state variable, fill from loaded state
+        current_state.camera.camera_matrix.release();
+        for(auto const &value : state_to_load.camera_matrix()){
+            current_state.camera.camera_matrix.push_back(value);
```

Comment:
I'm not too sure about using `push_back` here on a `cv::Mat`. It might be better to construct a new `cv::Mat` instance using `state_to_load.camera_matrix().data()` to get the underlying array. You could then reshape the matrix to be a proper 3x3 matrix and then clone it as you do in `values_by_comma_to_mat()`